pythonPackages.django_5: remove failing patch

phaer opened the pull request:
...because it was upstreamed and is included in 5.0.3, so that ´patch´ fails with
> Reversed (or previously applied) patch detected!

which means that django_5 does not build on current master. Tests do pass on 3.12 without this error.

Comments in https://github.com/NixOS/nixpkgs/pull/293227 suggests that another error during the test run might have shadowed this on.

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mweinelt commented:
Duplicate of #299212. But also, Django 5.0.3 is cached on cache.nixos.org, so it did built at the time I merged the PR.

```
$ git log --oneline pkgs/development/python-modules/django/5.nix
[...]
a4c561970fa9 python312Packages.django_5: 5.0.2 -> 5.0.3
$ nix-build -A python311Packages.django_5
[...]
copying path '/nix/store/li22yckwxslwx7agk1pkrn5563lbi6x4-python3.11-Django-5.0.3' from 'https://cache.nixos.org'
/nix/store/li22yckwxslwx7agk1pkrn5563lbi6x4-python3.11-Django-5.0.3
```

mweinelt commented:
Looks like a bad merge between python-updates (via staging-next) and master.

phaer commented:
Thanks for the quick response & sorry for the noise. Had somehow missed that PR in the files history.

phaer committed:
pythonPackages.django_5: remove failing patch...

...because it was upstreamed and is included in 5.0.3,
so that ´patch´ fails with
> Reversed (or previously applied) patch detected!
which means that django_5 does not build on current master.

Comments in https://github.com/NixOS/nixpkgs/pull/293227
suggests that another error during the test run might have
shadowed this on.
